docs/embedded_content.rst

```diff
@@ -0,0 +1,105 @@
+################
+Embedded Content
+################
+
+In ``insights-core``, we separate the return of the results from the rendering of
+the results.  This separation allows applications using ``insights-core``
+to produce output appropriate to their application.  So, for example,
+the Red Hat Insights product uses an approach that utilizes several
+content artifacts (general descriptions, specific descriptions, ansible
+playbooks, tagging, etc.) while an internal diagnostic system has a very
+simple description and tagging approach.
+
+Because of this separation, the methods that return results
+(``make_response``, ``make_pass``, and ``make_fail``) should not provide
+formatting themselves.  Instead, keyword arguments (kwargs) are used to
+pass back information from the plugin, to be interpolated by the caller.
+
+However, in the case of creating rules for individual use, structuring
+the data returned by the command line invocation of insights-core, the
+separation can add unneeded complexity.  For this reason, a conventional
+approach using a simple string or dictionary has been created.
+
+``CONTENT`` Attribute
+=====================
+
+To embedded content with a rule, create a ``CONTENT`` attribute.  This
+attribute can be a string or a dictionary.  When it is a string, it is
+interpreted as a jinja2 string.  The kwargs of the ``make_*`` functions
```

Comment:
Would it be useful to include a link to the jinga2 docs here for reference?

http://jinja.pocoo.org/docs/2.10/

Reply:
Ack, added.